Server/src/models/DatasetModels/DatasetApprovalModel.ts

```diff
@@ -0,0 +1,88 @@
+import { Connection, getConnection } from "typeorm";
+import { Dataset } from "../entities/Dataset";
+import { selectUnapprovedDatasetInfoQuery, Unapproveddatasets } from "../entities/Unapproveddatasets";
+import { IDatasetIDModel } from "../interfaces/DatasetModelInterface";
+import { DatasetCommonModel } from "./DatasetCommonModel";
+
+
+
+export class DatasetApprovalModel {
+    private connection: Connection;
+    private commonModel: DatasetCommonModel
+    constructor() {
+        this.connection = getConnection();
+        this.commonModel = new DatasetCommonModel();
+    }
+
+    private selectDatasetCommentQuery = (id: number) =>
+        this.connection.createQueryBuilder(Dataset, 'dataset')
+            .select('dataset.comments', 'comments')
+            .where('dataset.id = :id', { id: id })
+            .getRawOne();
+
+    async selectUserFlaggedDatasets(uploaderId: number): Promise<IDatasetIDModel[]> {
+        let userFlaggedDatasets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_Datasets')

```

Comment:
it is unapproved_Datasets in createQueryBuilder but unapproved_datasets in the select, is it fine?

Reply:
Will change. This will crash the code

Reply:
Will change. This will crash the code

Reply:
Fixed